reconcile/utils/saasherder.py

```diff
@@ -322,6 +328,22 @@ def _validate_upstream_not_used_with_commit_sha(
                     f'upstream used with commit sha: {ref}')
                 self.valid = False
 
+    def _validate_image_tag_not_equals_ref(
+            self,
+            saas_file_name: str,
+            resource_template_name: str,
+            ref: str,
+            parameters: dict,
+    ):
+        image_tag = parameters.get('IMAGE_TAG')
+        if image_tag and str(ref).startswith(str(image_tag)):
```

Comment:
Is a prefix match accurate here? Judging from function and PR description I'd assume to check for an exact match rather than a prefix, e.g.,

```
ref.strip() == image_tag.strip()
```

Reply:
Ah I see the use-case now in the test. All good.

Reply:
the IMAGE_TAG is the first <app-interface settings.hashLength>, so a prefix is what we need.